### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by mike-plummer**
Library upgrades
Adding lots more content re: basic and advanced JS, ecosystem, testing, typing

### CODE REVIEW

**Comment by DSchau** [comment]
Deploy preview for *javascript-workshop* ready!

Built with commit a3585100e38f47be395f4dc8d90069ab64c177f8

https://deploy-preview-4--javascript-workshop.netlify.com

**Comment by DSchau** [comment]
Deploy preview for *javascript-workshop* ready!

Built with commit 48b09da9cafeef84a8a8ff08eecb01af07423c17

https://deploy-preview-4--javascript-workshop.netlify.com

File: src/slides/03-advanced-js/index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Advanced JS
```
Comment (JakePartusch) [inline src/slides/03-advanced-js/index.md:1]: We could also add generator functions, but that might hurt more than help :)

File: src/slides/04-testing/index.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const func = jest.fn(); // Build a mock function
+  func(1, 2, 3);
+  expect(func).toHaveBeenCalledTimes(1);
+  expect(func).toHaveBeenCalledWith(1, 2, 3);
+});
+```
+
+---
+
+### Data-driven testing
```
Comment (JakePartusch) [inline src/slides/04-testing/index.md:56]: Might want to talk about `jest-each` if we are adding this section
https://www.npmjs.com/package/jest-each

**Review by JakePartusch — APPROVED** [review]

File: src/slides/03-advanced-js/index.md (hunk 1 above)
Comment (mike-plummer) [reply src/slides/03-advanced-js/index.md:1]: Yeah, consciously left them out at this point. Very rare to have to work with them, and by the time you're ready to write one you're well past the level of these slides

**Pull request merged**